docs/rules/no-wait-for-empty-callback.md

```diff
@@ -0,0 +1,40 @@
+# Empty callbacks inside `waitFor` and `waitForElementToBeRemoved` are not preferred (no-wait-for-empty-callback)
+
+## Rule Details
+
+This rule aims to ensure the correct usage of `waitFor` and `waitForElementToBeRemoved`, in the way that they're intended to be used.
+If an empty callback is passed, these methods will just wait next tick of the event loop before proceeding, and that's not consistent with the philosophy of the library.
+**Instead, insert an assertion in that callback function.**
```

Comment:
@timdeschryver I've updated the rule details with proper suggestion fix from dom-testing-library doc. I'm even thinking about [providing a suggestion](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions) for this rule!

Reply:
Mmm it needs to provide a suggestion fix too, not just the description, so we can't use that here.

Reply:
@Belco90 thanks and TIL eslint suggestions.
So if I understand correctly we can't provide a suggestion without a fix?

Reply:
I'm afraid so:

> The suggestion objects represent individual suggestions that could be applied and require either a desc key string that describes what applying the suggestion would do or a messageId key (see below), and a fix key that is a function defining the suggestion result.